content: draft: Update Source Track terms (#1471)
slsa-framework/slsa: docs/spec/draft/source-requirements.md

| Term | Description
| --- | ---

> | Organization | A set of people who collectively create source revisions within a Source Repository. Examples of organizations include open-source projects, a company, or a team within a company. The organization defines the goals of a Source Repository and the methods used to produce new revisions.

Author: Filed #1480 to track.

| Source Management Tool | Software for the creation and tracking of Source Revisions within a Source Repository, e.g. Git, Subversion.
| Source Control System (SCS) | The platform or combination of services (self-hosted or SaaS) that hosts a Source Repository and provides a trusted foundation for managing revisions by enforcing policies for authentication, authorization, and change management, such as mandatory code reviews or passing status checks.
| Source Provenance | Information about how a revision came to exist, where it was hosted, when it was generated, what process was used, who the contributors were, and which parent revisions preceded it.
| Source Repository (Repo) | A logical unit of files, their full revision history, as well as Branches, Tags, and other metadata.  Source Repositories are uniquely identified by a stable locator, such as a URI.

> | Source Revision | A specific, logically immutable snapshot of the repository's tracked files at a point in  time. It is uniquely identified by a revision identifier, such as a cryptographic hash like a Git commit SHA or a path-qualified sequential number like 25@trunk/ in SVN. A Source Revision MUST include both the content (the files) and its associated version control metadata, such as the author, timestamp, and parent revision(s). Source Revisions are regularly used as input to a build system (see [SLSA Build Track](build-requirements.md)). Note: Path qualification is needed for version control systems that use represent Branches and Tags using paths, such as Subversion and Perforce.

Author: The requirement that Source Revisions include author, timestamp, and parent revisions, should go a long way to help with #1460?

> | Source Revision | A specific, logically immutable snapshot of the repository's tracked files at a point in  time. It is uniquely identified by a revision identifier, such as a cryptographic hash like a Git commit SHA or a path-qualified sequential number like 25@trunk/ in SVN. A Source Revision MUST include both the content (the files) and its associated version control metadata, such as the author, timestamp, and parent revision(s). Source Revisions are regularly used as input to a build system (see [SLSA Build Track](build-requirements.md)). Note: Path qualification is needed for version control systems that use represent Branches and Tags using paths, such as Subversion and Perforce.

Author: So in this context (just describing what a Source Revision is) I don't think it's necessary for it to be trustworthy/unforgeabe?

Those things can get layered on later in the requirements section though?

> | Source Revision | A specific, logically immutable snapshot of the repository's tracked files at a point in  time. It is uniquely identified by a revision identifier, such as a cryptographic hash like a Git commit SHA or a path-qualified sequential number like 25@trunk/ in SVN. A Source Revision MUST include both the content (the files) and its associated version control metadata, such as the author, timestamp, and parent revision(s). Source Revisions are regularly used as input to a build system (see [SLSA Build Track](build-requirements.md)). Note: Path qualification is needed for version control systems that use represent Branches and Tags using paths, such as Subversion and Perforce.

Reviewer: > So in this context (just describing what a Source Revision is) I don't think it's necessary for it to be trustworthy/unforgeabe?

Yes, I agree. 
the VCS typically has _forgeable_ metadata and _trustworth_ version identifiers, higher tiers of SCS attempt to produce trustworthy metadata too, but this requires other tools if your VCS is distributed like git.

| Branch | An evolving named reference that points to a specific Source Revision within a Source Repository. It is expected to be updated over time to point to new Source Revisions as development occurs. Different branches may have different security requirements. For example: `main`, `develop`, `feature-x`.

> | Tag | A named reference to a specific Source Revision within a Source Repository. Tags are often used to to permanently mark a specific point in history, such as an official release version (e.g., `v1.2.3`, `release-20250722`), but may also be used to make a temporary assoiation with a Source Revision (e.g. `latest`, `nightly`).

Author: Let me provide a further tweak to that table

| Term | Description |
| --- | --- |
| **Named Reference** | A user-friendly name for a specific source revision, such as `main` or `v1.2.3`. |
| **Branch** | A *named reference* that points to a line of active development and is expected to be updated frequently.  E.g. `main`, `develop`, or `feature-x` |
| **Tag** | A *named reference* used to mark a specific source revision, often for a release.  Tags may be updated to point to different revisions but there is not necessarily continuity from one revision to another.  E.g. `v1.2.3`, `release-20250722`, `latest`, `nightly`. |
| **Static Tag** | A *tag* that is intended to be immutable. Once created, it is not moved to point to a different revision. E.g. `v1.2.3`, `release-20250722` |

> | Tag | A named reference to a specific Source Revision within a Source Repository. Tags are often used to to permanently mark a specific point in history, such as an official release version (e.g., `v1.2.3`, `release-20250722`), but may also be used to make a temporary assoiation with a Source Revision (e.g. `latest`, `nightly`).

Reviewer 2: I like "Named Reference" and "Static Tag". Can we drop "branch" and "tag" altogether and always use those first two terms?  "Branch" and "tag" just generate a lot of confusion because people think they know how it works but it might not agree with the definition. "Named Reference" and "Static Tag," on the other hand, sound like what we mean and do not have baggage of conflicting terms.

> | Tag | A named reference to a specific Source Revision within a Source Repository. Tags are often used to to permanently mark a specific point in history, such as an official release version (e.g., `v1.2.3`, `release-20250722`), but may also be used to make a temporary assoiation with a Source Revision (e.g. `latest`, `nightly`).

Author: Took another shot at this.

1. Using Named Reference, Branch, and Tag.
2. Not using '_static_ tag' because I realized the generic 'tag' was really just a 'named reference'.  We don't need them both.
3. Keeping 'branch' because it is an essential concept.  It's where the development occurs and it's what most of our other protections apply to.  'Named reference' isn't sufficient since it doesn't track development over time.  E.g. a branch is more than just the reference itself, it's all the changes that occurred in that context.  Does that come across in the new terminology?
5. Using the 'line of development' terminology that occurs in both [git](https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-branch) and [subversion](https://svnbook.red-bean.com/en/1.7/svn-book.html#svn.branchmerge.whatis) documentation when describing branches.
6. Added a NOTE as discussed in [the spec meeting](https://docs.google.com/document/d/1PwhekVB1iDpcgCQRNVN_aesoVdOiTruoebCs896aGxw/edit?tab=t.0#heading=h.knz5qwnil10g) to address some of the issues here.  Anything else I should add?

> | Change | A proposal to update a Specific Branch to a different Source Revision. A change can be proposed and reviewed before being accepted.

Reviewer: ```suggestion
| Change | A proposal to update a Named Reference to a different Source Revision. A change can be proposed and reviewed before being accepted.
```

> | Change | A Change is an atomic operation that updates a Branch or Tag to point from a prior Source Revision to a new Source Revision. A Change represents the unit of work that is proposed, reviewed, and accepted into the repository, and may encompass one or more individual commits (e.g., a merged pull request).

Reviewer: ```suggestion
| Change | A proposal to update a Specific Branch to a different Source Revision. A change can be proposed and reviewed before being accepted.
```

Here I think the previous text was slightly better?
There are multiple parts of a "change"
1. the uploaded changed files
2. the uploaded proposed new revision 
3. the proposed new state of a ref/branch/context

It's also possible to move a ref to an existing revision (such as a rollback or force push)

| Change History | A record of the history of revisions that preceded a specific revision.

> | Push / upload / publish | When an actor creates or modifies the Source Revisions, Branches or Tags in the repository.

Reviewer: ```suggestion
```

> | Review / approve / vote | When an actor uses a change management tool to comment upon, endorse, or reject a Change.

Reviewer: ```suggestion
```
